visidata/apps/vdsql/clickhouse.py

```diff
@@ -34,13 +34,13 @@ def iterload(self):
 
                 with Progress(gerund='clickhousing', sheet=self) as prog:
                     settings = {'max_block_size': 10000}
-                    self.query_result = con.con.execute_iter(sqlstr, settings, chunk_size=1000, query_id=qid)
-                    qid = str(time.time())
-                    for row in self.query_result:
-                        prog.total = con.con.last_query.progress.total_rows
-                        prog.made = con.con.last_query.progress.rows
+                    with con.con.query_rows_stream(sqlstr, settings) as stream:
+                        prog.total = int(stream.source.summary['total_rows_to_read'])
```

Comment:
For context, if I look at `stream.source.summary` when we enter this block after opening the `benchmark_results` table it looks like this:

```json
{
    "read_rows": "16384",
    "read_bytes": "360448",
    "written_rows": "0",
    "written_bytes": "0",
    "total_rows_to_read": "500",
    "result_rows": "0",
    "result_bytes": "0",
    "query_id": "c413bc6a-5bd7-40a5-81bc-a64ef53dc9c8"
}
```